app/src/main/java/com/example/ide/ui/viewmodel/MainViewModel.kt

```
@@ -1373,6 +1394,51 @@ Opciones de importación:
+    }
+
+    fun rejectPendingPatch() {
+        pushAssistantMessage("🛑 Patch Preview rechazado por usuario")
+    }
+
+    fun runTerminalCommand(command: String) {
+        val cmd = command.trim()
+        if (cmd.isBlank()) return
+        val allowed = setOf("pwd", "ls", "cd", "cat", "mkdir", "rm")
+        val first = cmd.substringBefore(" ")
+        if (first !in allowed || cmd.contains("..") || cmd.contains("../") || cmd.startsWith("/")) {
```

detekt (1.23.8)

[warning] This condition is too complex (4). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)